packages/projection/test/operators/Mappers/withHandles.test.ts

```diff
@@ -288,4 +292,58 @@ describe('withHandles', () => {
       expect(handles[0].latestOwnerAddress).toBe(maryAddress);
     });
   });
+
+  describe('subhandles', () => {
+    const project = (tx: Cardano.OnChainTx) =>
+      firstValueFrom(
+        of({
+          block: {
+            body: [tx],
+            header: mockProviders.ledgerTip
+          }
+        } as ProjectionEvent).pipe(withUtxo(), withCIP67(), withHandles({ policyIds: [handlePolicyId] }, logger))
+      );
+
+    it('adds parentHandle data for virtual subhandles', async () => {
+      const { handles } = await project({
+        body: { outputs: [virtualSubHandleOutput] },
+        inputSource: Cardano.InputSource.inputs
+      } as Cardano.OnChainTx);
+
+      expect(handles).toHaveLength(1);
+
+      expect(handles[0].latestOwnerAddress).toBe(
+        'addr_test1qpadxfxylvy8p8wejlmt9wnesr2squgr524r77n7hz6yh3h34r3hjynmsy2cxpc04a6dkqxcsr29qfl7v9cmrd5mm89qqh563f'
+      );
+      expect(handles[0].parentHandle).toBe('handl');
+      expect(handles[0].handle).toBe('virtual@handl');
+    });
+
+    it('adds parentHandle data for NFT subhandles', async () => {
+      const { handles } = await project({
+        body: { outputs: [NFTSubHandleOutput] },
+        inputSource: Cardano.InputSource.inputs
+      } as Cardano.OnChainTx);
+
+      expect(handles).toHaveLength(1);
+
+      expect(handles[0].parentHandle).toBe('handl');
+      expect(handles[0].handle).toBe('sub@handl');
+    });
+
+    it('includes a handle with "null" address, when transaction burns a subhandle', async () => {
+      const { handles } = await project({
+        // TODO: Fix this to construct a proper CIP67 asset and not use `assetIdFromHandle`
+        body: {
+          mint: new Map([[Cardano.AssetId.fromParts(handlePolicyId, subhandleAssetName), -1n]]),
+          outputs: [NFTSubHandleOutput] as Cardano.TxOut[]
```

Comment:
if you don't specify any outuputs then `withUtxo(), withCIP67()` never specifies or sets the cip67 assets, so it never parses the handles because they are invalid UTF8handles, [line](https://github.com/input-output-hk/cardano-js-sdk/blob/b2fc531cece813e1ea89db59d3ab90cfa6204bc2/packages/projection/src/operators/Mappers/withHandles.ts#L41).

Reply:
That sounds like a bug to me, exactly the reason for this change in test :)

Reply:
addressed here: 629801a081171debd6ddbd070c2cf01e8a0ea0ce  name improved here: e8d7fea3fd87c8826f4f82d426d6d853cc8f2fb9
We don't use the `cip67Asset` here since burned handles don't have a `cip67Asset` set so I decode it
manually here to see if it is a cip67 asse

Reply:
The purpose of `withCIP67` is to parse it just once, then other mappers can look up assets. I suggest extending `withCIP67` to include minted assets instead of duplicating the parser

Reply:
I wanted to do something like this but it seemed like a larger change. The current implementation of `withCIP67` [here](https://github.com/input-output-hk/cardano-js-sdk/blob/171196916244d0bcde83b18d509669c2c38a0d63/packages/projection/src/operators/Mappers/withCIP67.ts#L25) works with the data produced by `withUtxo` [here](https://github.com/input-output-hk/cardano-js-sdk/blob/39cbabbe4cc7b13e6447d1eafaa22ad880fd348a/packages/projection/src/operators/Mappers/withUtxo.ts#L17) which assumes there is data in the `outputs` array. 

Are you suggesting I modify the `withUtxo` function, or should there be some other mapper that also adds data from the `mint` entries? It didn't make sense to me when I originally looked at it to modify the `withUtxo` since we don't actually have a Utxo to work with and the function's purpose wouldn't be as clear anymore.

Reply:
No, if you added it to withUtxo it wouldn't be correct. You can add a `WithMint` dependency to `withCip67` mapper

Reply:
should be addressed here: 461a2a90b5e16ca98d4b96448256d0506b930011